next.config.js

```diff
@@ -245,6 +245,40 @@ const defaultConfig = {
         destination: '/docs/changelog',
         permanent: false,
       },
+      {
+        source: '/',
+        has: [
+          { type: 'cookie', key: 'neon_login_indicator' },
+          { type: 'header', key: 'referer' },
+        ],
+        destination: '/home',
+        permanent: false,
+      },
+      {
+        source: '/',
+        has: [
+          { type: 'cookie', key: 'neon_login_indicator' },
+          { type: 'header', key: 'referrer' },
+        ],
+        destination: '/home',
+        permanent: false,
+      },
+      {
+        source: '/',
+        has: [{ type: 'cookie', key: 'neon_login_indicator' }],
+        missing: [
+          { type: 'header', key: 'referer' },
+          { type: 'header', key: 'referrer' },
+        ],
+        destination: 'https://console.neon.tech',
```

Comment:
afaik it would be the same as running those in the middleware.js file, since the path matching and redirects config are invoked before cache is served.